test/e2e/framework/service_util.go

```diff
@@ -1450,3 +1461,28 @@ func GetServiceLoadBalancerCreationTimeout(cs clientset.Interface) time.Duration
 	}
 	return LoadBalancerCreateTimeoutDefault
 }
+
+func TestAffinityFromPod(execPod *v1.Pod, targetIpPort string) bool {
+	var hostname string
+	cmd := fmt.Sprintf(`wget -qO- http://%s/ -T 1`, targetIpPort)
+	affinityCount := 0
+	if pollErr := wait.PollImmediate(Poll, ServiceTestTimeout, func() (bool, error) {
+		stdout, err := RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		if err != nil || len(stdout) == 0 {
+			Logf("Failed to get response from %s. Retry until timeout", targetIpPort)
+			return false, nil
+		}
+		Logf("Got pod response: %s", stdout)
+		if hostname == stdout {
+			affinityCount++
+		} else {
+			affinityCount = 0
+			hostname = stdout
+		}
+		return affinityCount >= AffinityConfirmCount, nil
+	}); pollErr != nil {
+		// Affinity check timed out.
+		return false
```

Comment:
This function mixes these two cases:
- Pod is able to get response from backends but affinity doesn't hold (return false).
- Pod is unable to get response from backends (return false).

Is it worth to return (boolean, error) to help distinguishing different cases?

Reply:
I think not. I put "Affinity shouldn't hold or connection timeout." as the error message. And if it is connection timeout, there must be something really bad happened.